lib/relation-definition.js

```diff
@@ -532,37 +532,85 @@ function lookupModel(models, modelName) {
   }
 }
 
-function lookupModelTo(modelFrom, modelTo, params, singularize) {
-  if ('string' === typeof modelTo) {
+/*
+ * @return {Object} modelFrom Instance of the 'from' model
+ * @param {Object|String} modelToRef Reference to Model object to which you are
+ *  creating the relation: model instance, model name, or name of relation to model.
+ * @param {Object} params relation params
+ * @param {Boolean} singularize Whether the modelToRef should be singularized when
+ *  looking-up modelTo
+ * @return {Object} modelTo Instance of the 'to' model
+ */
+function lookupModelTo(modelFrom, modelToRef, params, singularize) {
+  var modelTo;
+
+  if ('string' === typeof modelToRef) {
     var modelToName;
-    params.as = params.as || modelTo;
-    modelTo = params.model || modelTo;
-    if (typeof modelTo === 'string') {
-      modelToName = (singularize ? i8n.singularize(modelTo) : modelTo).toLowerCase();
-      modelTo = lookupModel(modelFrom.dataSource.modelBuilder.models, modelToName) || modelTo;
-    }
+    modelTo = params.model || modelToRef; // modelToRef is modelTo name
+    params.as = params.as || modelToRef;  // modelToRef is relationTo name
+
     if (typeof modelTo === 'string') {
-      modelToName = (singularize ? i8n.singularize(params.as) : params.as).toLowerCase();
-      modelTo = lookupModel(modelFrom.dataSource.modelBuilder.models, modelToName) || modelTo;
+      // lookup modelTo by name
+      modelToName = modelTo;
+      modelToName = (singularize ? i8n.singularize(modelToName) : modelToName).toLowerCase();
+      modelTo = lookupModel(modelFrom.dataSource.modelBuilder.models, modelToName);
     }
-    if (typeof modelTo !== 'function') {
-      throw new Error(g.f('Could not find "%s" relation for %s', params.as, modelFrom.modelName));
+    if (!modelTo) {
+      // lookup modelTo by relationTo name
+      var relationToName = params.as;
+      modelToName = (singularize ? i8n.singularize(relationToName) : relationToName).toLowerCase();
+      modelTo = lookupModel(modelFrom.dataSource.modelBuilder.models, modelToName);
     }
+  } else {
+    // modelToRef might already be an instance of model
+    modelTo = modelToRef;
+  }
+  if (typeof modelTo !== 'function') {
+    throw new Error(g.f('Could not find relation %s for model %s', params.as, modelFrom.modelName));
```

Comment:
i asked myself this one too, and did a check everywhere `lookUpModelTo()` is used : if it were to return null, there would be plenty of `property x of undefined`errors. I feel the code should never return null and throw instead, hence the fix